### PR TITLE
handle warnings from reselect about broken selectors rules

### LINKED ISSUE #47087: Get rid of selectors warnings from browser console

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/47087

### Description

Cleanup warnings in console about wrong usage of selectors. Some of the warnings are valid, some are false positives, so we remove noise from them

One warning is valid, and I'll keep it to handle properly later: `const getQuestionsMemoized = createDeepEqualSelector`

### How to verify

- open app in dev mode
- verify console

### Before

https://github.com/user-attachments/assets/d714a1d4-b57e-4c83-a3fb-09d5632db6e7

### After

https://github.com/user-attachments/assets/6a7f1efb-7371-4942-b777-11c950be6976

### CODE REVIEW

File: frontend/src/metabase/lib/entities.js [hunk 1]
```diff
@@ -465,18 +465,22 @@ export function createEntity(def) {
   const getLoading = createSelector(
     [getRequestState],
     requestState => requestState.loading,
+    { devModeChecks: { inputStabilityCheck: "never" } },
```
Comment (kamilmielnik) [inline frontend/src/metabase/lib/entities.js:468]: :x: These 4 warnings are correct and can be fixed by introducing a constant for the empty object returned in line 463.

File: frontend/src/metabase/query_builder/components/QueryVisualization.jsx [hunk 2]
```diff
@@ -104,7 +104,11 @@ export const VisualizationEmptyState = ({ className }) => (
 export function VisualizationRunningState({ className = "" }) {
   const [isSlow] = useTimeout(SLOW_MESSAGE_TIMEOUT);
 
-  const getLoadingMessage = useSelector(getWhiteLabeledLoadingMessageFactory);
+  const getLoadingMessage = useSelector(getWhiteLabeledLoadingMessageFactory, {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/QueryVisualization.jsx:107]: :x: This warning is correct and can be fixed by making sure [`getLoadingMessageFactory` returns the same function](https://github.com/metabase/metabase/blob/d6d2edb3e54530c0d871d0a2a7060361a2c80f80/frontend/src/metabase/plugins/index.ts#L213) every time.

File: frontend/src/metabase/query_builder/selectors.js [hunk 3]
```diff
@@ -276,6 +276,7 @@ export const getTables = createSelector(
 
     return [];
   },
+  { devModeChecks: { inputStabilityCheck: "never" } },
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/selectors.js:279]: :x: This warning is correct and can be fixed by returning the same empty array in line 277.

File: frontend/src/metabase/query_builder/selectors.js [hunk 4]
```diff
@@ -298,6 +299,7 @@ export const getSampleDatabaseId = createSelector(
     const sampleDatabase = _.findWhere(databases, { is_sample: true });
     return sampleDatabase && sampleDatabase.id;
   },
+  { devModeChecks: { inputStabilityCheck: "never" } },
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/selectors.js:302]: :x: This warning is correct and can be fixed by returning the same empty array in line 265.

**Review by kamilmielnik — CHANGES_REQUESTED** [review]
I'm ok with suppressing the warnings in `EntityListLoader` and `EntityObjectLoader` but the rest are true positives and can be easily fixed

**Comment by uladzimirdev** [comment]
@kamilmielnik thanks for review, valid points, everything should be fixed now

**Review by kamilmielnik — APPROVED** [review]
Many thanks :raised_hands: 

MLv2 warnings next? :smile:

**Comment by uladzimirdev** [comment]
@kamilmielnik unfortunately it looks like MLv2 warnings can be fixed only in clojure

**Pull request merged**